cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerServiceLauncher.java

```diff
@@ -168,16 +170,15 @@ public void run() {
             int diskSize = cConf.getInt(Constants.TaskWorker.CONTAINER_DISK_SIZE_GB);
             twillPreparer = ((StatefulTwillPreparer) twillPreparer)
               .withStatefulRunnable(TaskWorkerTwillRunnable.class.getSimpleName(), false,
-                                    new StatefulDisk("task-worker-data", diskSize,
-                                                     cConf.get(Constants.CFG_LOCAL_DATA_DIR)));
+                                    new StatefulDisk(STATEFUL_DISK_NAME, diskSize,
+                                                     cConf.get(Constants.CFG_LOCAL_DATA_DIR)))
+              .withReadonlyDisk(TaskWorkerTwillRunnable.class.getSimpleName(), STATEFUL_DISK_NAME);
```

Comment:
nit: Maybe it would make more sense to have readOnly as a property of the StatefulDisk instead of having a separate method in the interface to expose this functionality?

Reply:
It does. The problem is that right now one defines StatefulDisk for the main TwillRunnable (i.e., container), and it applies to all runnables (including sidecars) in the Twill application. The limitation stems from (1) orderedStart parameter which is pod specific, and (2) if we make statefuldisk definition per Runnable, then one needs to define stateful disks for all runnables inside a Twill application.